ble_server.characteristic.set_value broken on 2025.10.5=>2025.11.0

### The problem

Between esphome 2025.10.5 and 2025.11.0 the way values are initialized in ble_server was changed.
Somehow it seems the `set_buffer` function can not be found now, because of some problem with the template.

This broke all my existing yaml configs that use ble_server.

Compile error and demonstration yaml provided below

### Which version of ESPHome has the issue?

2025.11.0

### What type of installation are you using?

pip

### What platform are you using?

ESP32

### Component causing the issue

ble_server

### YAML Config

```yaml
#===============================================================================
substitutions:
  device_name: bletest
  ble_service_uuid: "0xce10"
#===============================================================================

esphome:
  name: ${device_name}
  on_boot:
    - script.execute: WriteBLESererChar

esp32:
  board: esp32-s3-devkitc-1
  framework:
    type: esp-idf
    # type: arduino

logger:

esp32_ble:
  name: "test"

esp32_ble_server:
  manufacturer: ""
  model: ""
  id: ble_server
  services:
    - uuid: ${ble_service_uuid}
      advertise: true
      characteristics:
        - uuid: "0xce82"
          id: ble_characteristic_test
          notify: true
          read: false
          write: false
          value:
            data: [0x00]
          indicate: false

psram:
   mode: octal # quad
   speed: 80MHz
   enable_ecc: true

script:
  - id: "WriteBLESererChar"
    mode: single
    then:
      - delay: 2s
      - ble_server.characteristic.set_value:
          id: ble_characteristic_test
          value:
            data: [0xfc, 0xef, 0xfe, 0x86]
      - ble_server.characteristic.notify:
          id: ble_characteristic_test
```

### Anything in the logs that might be useful for us?

```txt
src/main.cpp: In function 'void setup()':                         
src/main.cpp:284:97: error: no matching function for call to 'esphome::esp32_ble_server::esp32_ble_server_automations::BLECharacteristicSetValueAction<>::set_buffer(<brace-enclosed initializer list>)'
  284 |   esp32_ble_server_esp32_ble_server_automations_blecharacteristicsetvalueaction_id_5->set_buffer({252, 239, 254, 134});
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                        
In file included from src/esphome/components/esp32_ble/ble.h:14,                                                                                                           
                 from src/esphome.h:6,                                                                                                                                                                                                                                                                                                                
                 from src/main.cpp:3:
src/esphome/core/automation.h:24:29: note: candidate: 'template<class V> void esphome::esp32_ble_server::esp32_ble_server_automations::BLECharacteristicSetValueAction<Ts>::set_buffer(V) [with Ts = {}]'
   24 |   template<typename V> void set_##name(V name) { this->name##_ = name; }
      |                             ^~~~
src/esphome/core/automation.h:26:39: note: in expansion of macro 'TEMPLATABLE_VALUE_'
   26 | #define TEMPLATABLE_VALUE(type, name) TEMPLATABLE_VALUE_(type, name)
      |                                       ^~~~~~~~~~~~~~~~~~
src/esphome/components/esp32_ble_server/ble_server_automations.h:72:3: note: in expansion of macro 'TEMPLATABLE_VALUE'
   72 |   TEMPLATABLE_VALUE(std::vector<uint8_t>, buffer)
      |   ^~~~~~~~~~~~~~~~~
src/esphome/core/automation.h:24:29: note:   template argument deduction/substitution failed:
   24 |   template<typename V> void set_##name(V name) { this->name##_ = name; }
      |                             ^~~~
src/esphome/core/automation.h:26:39: note: in expansion of macro 'TEMPLATABLE_VALUE_'
   26 | #define TEMPLATABLE_VALUE(type, name) TEMPLATABLE_VALUE_(type, name)
      |                                       ^~~~~~~~~~~~~~~~~~
src/esphome/components/esp32_ble_server/ble_server_automations.h:72:3: note: in expansion of macro 'TEMPLATABLE_VALUE'
   72 |   TEMPLATABLE_VALUE(std::vector<uint8_t>, buffer)
      |   ^~~~~~~~~~~~~~~~~
src/main.cpp:284:97: note:   couldn't deduce template parameter 'V'
  284 |   esp32_ble_server_esp32_ble_server_automations_blecharacteristicsetvalueaction_id_5->set_buffer({252, 239, 254, 134});
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
In file included from src/esphome.h:15:
src/esphome/components/esp32_ble_server/ble_server_automations.h:73:8: note: candidate: 'void esphome::esp32_ble_server::esp32_ble_server_automations::BLECharacteristicSetValueAction<Ts>::set_buffer(esphome::bytebuffer::ByteBuffer) [with Ts = {}]'
   73 |   void set_buffer(ByteBuffer buffer) { this->set_buffer(buffer.get_data()); }
      |        ^~~~~~~~~~
src/esphome/components/esp32_ble_server/ble_server_automations.h:73:30: note:   no known conversion for argument 1 from '<brace-enclosed initializer list>' to 'esphome::bytebuffer::ByteBuffer'
   73 |   void set_buffer(ByteBuffer buffer) { this->set_buffer(buffer.get_data()); }
      |                   ~~~~~~~~~~~^~~~~~
```

### Additional information

_No response_

--- a/esphome/components/esp32_ble_server/ble_server_automations.h
+++ b/esphome/components/esp32_ble_server/ble_server_automations.h
@@ -70,6 +70,7 @@ template<typename... Ts> class BLECharacteristicSetValueAction : public Action<T
  public:
   BLECharacteristicSetValueAction(BLECharacteristic *characteristic) : parent_(characteristic) {}
   TEMPLATABLE_VALUE(std::vector<uint8_t>, buffer)
+  void set_buffer(std::initializer_list<uint8_t> buffer) { this->buffer_ = std::vector<uint8_t>(buffer); }
   void set_buffer(ByteBuffer buffer) { this->set_buffer(buffer.get_data()); }
   void play(const Ts &...x) override {
     // If the listener is already set, do nothing
@@ -115,6 +116,7 @@ template<typename... Ts> class BLEDescriptorSetValueAction : public Action<Ts...
  public:
   BLEDescriptorSetValueAction(BLEDescriptor *descriptor) : parent_(descriptor) {}
   TEMPLATABLE_VALUE(std::vector<uint8_t>, buffer)
+  void set_buffer(std::initializer_list<uint8_t> buffer) { this->buffer_ = std::vector<uint8_t>(buffer); }
   void set_buffer(ByteBuffer buffer) { this->set_buffer(buffer.get_data()); }
   void play(const Ts &...x) override { this->parent_->set_value(this->buffer_.value(x...)); }
 

--- a/tests/components/esp32_ble_server/common.yaml
+++ b/tests/components/esp32_ble_server/common.yaml
@@ -69,3 +69,11 @@ esp32_ble_server:
               - ble_server.descriptor.set_value:
                   id: test_change_descriptor
                   value: !lambda return bytebuffer::ByteBuffer::wrap({0x03, 0x04, 0x05}).get_data();
+              - ble_server.characteristic.set_value:
+                  id: test_change_characteristic
+                  value:
+                    data: [0xfc, 0xef, 0xfe, 0x86]
+              - ble_server.descriptor.set_value:
+                  id: test_change_descriptor
+                  value:
+                    data: [0x01, 0x02, 0x03]

